pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1515,6 +1515,7 @@ public static WorkerConfig initializeWorkerConfigFromBrokerConfig(ServiceConfigu
         String hostname = ServiceConfigurationUtils.getDefaultOrConfiguredAddress(
                 ServiceConfigurationUtils.getAppliedAdvertisedAddress(brokerConfig));
         workerConfig.setWorkerHostname(hostname);
+        workerConfig.setPulsarFunctionsCluster(brokerConfig.getClusterName());
```

Comment:
what about  setting this `brokerConfig.getClusterName()` value only if the user did not configure another value ? 
otherwise this change may introduce a behaviour change

Reply:
If user configure another value different with brokerConfig.getClusterName() , broker will fail to start, refer to #2328.
To avoid this to happen, we could force the value same with brokerConfig.getClusterName().